x-pack/plugins/rule_registry/common/schemas/8.0.0/index.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Values } from '@kbn/utility-types';
+import {
+  ALERT_INSTANCE_ID,
+  ALERT_UUID,
+  ALERT_RULE_CATEGORY,
+  ALERT_RULE_CONSUMER,
+  ALERT_RULE_EXECUTION_UUID,
+  ALERT_RULE_NAME,
+  ALERT_RULE_PRODUCER,
+  ALERT_RULE_TYPE_ID,
+  ALERT_RULE_UUID,
+  SPACE_IDS,
+  ALERT_RULE_TAGS,
+  TIMESTAMP,
+} from '@kbn/rule-data-utils';
+
+/* DO NOT MODIFY THIS SCHEMA TO ADD NEW FIELDS. These types represent the alerts that shipped in 8.0.0.
+Any changes to these types should be bug fixes so the types more accurately represent the alerts from 8.0.0.
+
+If you are adding new fields for a new release of Kibana, create a new sibling folder to this one
+for the version to be released and add the field(s) to the schema in that folder.
+
+Then, update `../index.ts` to import from the new folder that has the latest schemas and add the
+new schemas to the union of all alert schemas.
+*/
+
+const commonAlertFieldNames = [
+  ALERT_RULE_CATEGORY,
+  ALERT_RULE_CONSUMER,
+  ALERT_RULE_EXECUTION_UUID,
+  ALERT_RULE_NAME,
+  ALERT_RULE_PRODUCER,
+  ALERT_RULE_TYPE_ID,
+  ALERT_RULE_UUID,
+  SPACE_IDS,
+  ALERT_RULE_TAGS,
+  TIMESTAMP,
+];
+export type CommonAlertFieldName800 = Values<typeof commonAlertFieldNames>;
```

Comment:
I agree that ending with `800` is a bit awkward. I initially wanted to use something more like `CommonAlertFieldName_8_0_0` at first, but the naming convention linter complains about using underscores if the type isn't all caps. Alerting uses a similar convention in the [SO migrations logic](https://github.com/elastic/kibana/blob/8.0/x-pack/plugins/alerting/server/saved_objects/migrations.ts#L97) which seems to have worked so far so I adopted that instead of trying to disable the linter rule for these type names.

> If we are going to bump it by one with each change then we probably don't need to start with 800

With this proposal we wouldn't bump the number by 1, instead we'd increment it to the version of Kibana that's shipping the change, e.g. the next one might be `CommonAlertFieldName830` for 8.3.0 if we make changes soon. But we haven't made changes in 8.1 or 8.2 so there would never be a `CommonAlertFieldName810` or `820`.

In the past we used integer version numbers, starting at `1` and incrementing on each Kibana version that shipped changes, for the legacy signals mappings and it was pretty confusing IMO. Most of the time when looking at version numbers what I wanted to now was what version of Kibana a particular version number shipped with so it would have saved time to just version the signals mappings with the Kibana version. Instead with the integer it has to be cross-referenced with Git history to figure out when it shipped.

The other possible option I considered was dropping the `800` from the name entirely and relying on the `8.0.0` in the folder path to differentiate this `CommonAlertFieldName` type from possible future `CommonAlertFieldName` types. I worried that that would make it too easy to accidentally import the wrong type though.